### PR TITLE
Grant access to `mysql` database when in dolt shell

### PR DESCRIPTION
**Pull request by JCOR11599**
Creating a new SQL engine will load privileges by default.

### CODE REVIEW

File: go/cmd/dolt/commands/sql.go [hunk 1]
```diff
@@ -377,12 +379,25 @@ func execShell(
...
+	}}
+
+	se, err := engine.NewSqlEngine(ctx, mrEnv, format, initialDb, false, tempUsers, true)
 	if err != nil {
 		return errhand.VerboseErrorFromError(err)
 	}
 	defer se.Close()
 
+	// Load MySQL db
+	se.GetUnderlyingEngine().Analyzer.Catalog.GrantTables.SetPersistCallback(privileges.SavePrivileges)
```
Comment (zachmu) [inline go/cmd/dolt/commands/sql.go:395]: All of this setup code should live in NewEngine

File: integration-tests/bats/sql-shell.bats [hunk 2]
```diff
@@ -42,6 +42,13 @@ teardown() {
     [[ "$output" =~ "+---------------------" ]] || false
 }
 
+@test "sql-shell: dolt sql shell has mysql db" {
```
Comment (zachmu) [inline integration-tests/bats/sql-shell.bats:45]: Need tests of adding grants via `-q`, listing them again

**Review by zachmu — COMMENTED** [review]
Good start but I want to consolidate this setup code into one place

File: integration-tests/bats/sql-shell.bats (hunk 2 above)
Comment (JCOR11599) [reply integration-tests/bats/sql-shell.bats:45]: Tests for `sql -q` are in `integration-tests/bats/sql.bats`, but they won't persist right now. They will after my refactor changes though.

File: go/cmd/dolt/commands/engine/sqlengine.go [hunk 3]
```diff
@@ -53,7 +55,10 @@ func NewSqlEngine(
 	format PrintResultFormat,
 	initialDb string,
 	isReadOnly bool,
-	tempUsers []gms.TemporaryUser,
+	mysqlDbFilePath string,
```
Comment (zachmu) [inline go/cmd/dolt/commands/engine/sqlengine.go:58]: This method now officially has too many params. Time to introduce a config struct to hold them.

You can do that on a second pass after you get this in

File: go/libraries/doltcore/sqle/mysql_file_handler/file_handler.go [hunk 4]
```diff
@@ -25,6 +25,8 @@ import (
 	"github.com/dolthub/go-mysql-server/sql/mysql_db"
 )
 
+const defaultMySQLFilePath = "mysql.db"
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/mysql_file_handler/file_handler.go:28]: This should go in a .dolt directory

Do that in a follow-up

**Review by zachmu — APPROVED** [review]
LGTM

**Pull request merged**